app/forms/new_phone_form.rb

```diff
@@ -88,6 +89,13 @@ def validate_not_voip
     end
   end
 
+  def validate_not_duplicate
+      current_user_phones = user.phone_configurations.map(&:phone)
+
+      return unless current_user_phones.include?(phone)
+      errors.add(:phone, I18n.t('errors.messages.phone_duplicate'))
```

Comment:
Can we make sure to normalize the phone numbers?

I want to say sometimes we store phone numbers formatted `(XXX) XXX-XXXX` and sometimes like `+1XXXXXXXX` etc etc 

I think a clear way normalize would be to make sure everything is in `e164` format, and the `Phonelib` gem does that for us. Here's a prior example of normalizing: https://github.com/18F/identity-idp/blob/master/lib/feature_management.rb#L148-L149

```suggestion
      current_user_phones = user.phone_configurations.map { |config| Phonelib.parse(config).e164 }

      return unless current_user_phones.include?(parsed_phone.e164)
      errors.add(:phone, I18n.t('errors.messages.phone_duplicate'))
```

Reply:
The `phone` value that we are comparing against is formatted with the `PhoneFormatter` service. If we're going to format the phones, I'd suggest using that:

```
current_user_phones = user.phone_configurations.map { |config| PhoneFormatter.format(config.phone) }
```

The phone _should_ be formatted before being written, but agree it is better to fomat them here to be safe.

Reply:
I just pushed a commit that involved removing an old test that conflicted with this one. PTAL. The commit formats both phone numbers before checking for duplicates.